woocommerce/woocommerce
# WooCommerce Docs: Support Gutenberg block conversion with CommonMark, add some basic unit tests.

### Changes proposed in this Pull Request:

This will close https://github.com/woocommerce/woocommerce/issues/38376

This adds support for basic Gutenberg blocks conversion from Markdown. We do it in 2 passes to keep things simple first we convert to HTML then we add Gutenberg block tags. (incidentally this is how its done in the copy/paste support in Gutenberg)

I also felt now is a good time to add unit tests setup, I haven't added it to CI yet, but the unit tests work with wp-env and PHPUnit. I haven't tested everything right away, it is my intent in some later follow up PRs to test some of the more integration-test kinda stuff like the job that processes manifests, but I expect some of those pieces of code to change more and be testing more high level behaviour so I think there is no rush to test them just yet.





### How to test the changes in this Pull Request:

1. Follow the test instructions from https://github.com/woocommerce/woocommerce/pull/38874
2. Go to edit one of the generated posts, open code-editor, see that gutenberg block tags were added surrounding elements in the markdown.
3. Also to run the tests locally and see those working run: `pnpm run test:unit` in the woocommerce docs plugin dir

### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
